pkg/cloud/cloud.go

```diff
@@ -320,9 +322,12 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 		}
 	}
 
+	// We hash the volume name to generate a unique token that is less than or equal to 64 characters
+	clientToken := sha256.Sum256([]byte(volumeName))
```

Comment:
I think encoding is enough here, why do we need extra hashing?

Reply:
in this step, we are hashing the `volumeName`, resulting in a byte slice of length 32 (256 bits). in the next step, we hex encode the slice and convert it to a string, ensuring it is always length 64. i think this is the safest way to guarantee we never exceed 64 characters in the token.